src/Config/Builder/EnumOptionBuilder.php

```diff
@@ -5,39 +5,72 @@
 namespace Phpcq\Config\Builder;
 
 use Phpcq\Config\Validation\Validator;
+use Phpcq\Exception\RuntimeException;
 use Phpcq\PluginApi\Version10\Configuration\Builder\EnumOptionBuilderInterface;
+use Phpcq\PluginApi\Version10\Configuration\Builder\FloatOptionBuilderInterface;
+use Phpcq\PluginApi\Version10\Configuration\Builder\IntOptionBuilderInterface;
+use Phpcq\PluginApi\Version10\Configuration\Builder\StringOptionBuilderInterface;
 
 final class EnumOptionBuilder extends AbstractOptionBuilder implements EnumOptionBuilderInterface
 {
     use TypeTrait;
 
-    public function ofStringValues(string ...$values) : EnumOptionBuilderInterface
+    /**
+     * @psalm-suppress PropertyNotSetInConstructor selfValidate() checks it.
+     * @var ConfigOptionBuilderInterface
+     */
+    private $valueBuilder;
+
+    public function ofStringValues(string ...$values): StringOptionBuilderInterface
     {
         $this->declareType('string');
         $this
             ->withValidator(Validator::stringValidator())
             ->withValidator(Validator::enumValidator($values));
 
-        return $this;
+        return $this->valueBuilder = new StringOptionBuilder($this->name, $this->description);
     }
 
-    public function ofIntValues(int ...$values) : EnumOptionBuilderInterface
+    public function ofIntValues(int ...$values): IntOptionBuilderInterface
     {
         $this->declareType('int');
         $this
             ->withValidator(Validator::intValidator())
             ->withValidator(Validator::enumValidator($values));
 
-        return $this;
+        return $this->valueBuilder = new IntOptionBuilder($this->name, $this->description);
     }
 
-    public function ofFloatValues(float ...$values) : EnumOptionBuilderInterface
+    public function ofFloatValues(float ...$values): FloatOptionBuilderInterface
     {
         $this->declareType('float');
         $this
             ->withValidator(Validator::floatValidator())
             ->withValidator(Validator::enumValidator($values));
 
-        return $this;
+        return $this->valueBuilder = new FloatOptionBuilder($this->name, $this->description);
+    }
+
+    public function normalizeValue($raw)
+    {
+        /** @psalm-suppress MixedAssignment */
+        $raw = $this->valueBuilder->normalizeValue($raw);
+
+        return parent::normalizeValue($raw);
+    }
+
+    public function validateValue($value): void
+    {
+        parent::validateValue($value);
+
+        $this->valueBuilder->normalizeValue($value);
+    }
+
+    public function selfValidate(): void
+    {
+        /** @psalm-suppress DocblockTypeContradiction */
+        if (null === $this->valueBuilder) {
```

Comment:
Will throw error that property has to be set prior retrieval in PHP 7.4 with typed properties, therefore we might want to change this to either explicitely `isset()` (preferred) here or initialize the property with `null` in it's declaration.

Reply:
But only if the define a property type, right?

Reply:
Correct, but we will then only have to adjust the declaration in the future when moving to typed properties instead of both locations.